### DISCUSSION TITLE
Customise Time-Grid's Sidebar

### DISCUSSION THREAD

**Discussion by Kay2dan** [vkurko/calendar#287] [Q&A]
Is it possible to customise the Sidebar in Time-grid with custom layout/ additional features?

![image](https://github.com/vkurko/calendar/assets/6276091/f7295395-f366-4ad2-868e-fb6b78ed6bb2)

_Use-case:_

I want to display user's local sun-rise & sun-set along-side the hours of the day. 
I may want to show histogram like graph of the sun-light over the day.

As strange as this request it, even a pointer towards how to extend a fork of the repo would be amazingly helpful.

Thank you

**Comment by vkurko** [comment]
Well, there is [slotLabelFormat](https://github.com/vkurko/calendar?tab=readme-ov-file#slotlabelformat) option which accepts a function as a value. The function can return arbitrary [content](https://github.com/vkurko/calendar?tab=readme-ov-file#content), for example, in the form of html. Perhaps this way you can insert the content you need. Unfortunately, there is no extended option for customizing the sidebar.

**Reply by Kay2dan** [reply]
Thank you, this is helpful, I will be looking into this soon. I will get back here after going over the docs & trying this out.

And thank you for an a lean Calendar package!
